src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,45 @@
 package core.basesyntax.service;
 
+import core.basesyntax.RegistrationException;
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int LENGTH_OF_LOGIN_AND_PASSWORD = 6;
+    private static final int USERS_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User cannot be null");
+        }
+
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login cannot be null or empty");
+        } else if (user.getPassword() == null) {
+            throw new RegistrationException("Password cannot be null or empty");
+        } else if (user.getAge() == null) {
+            throw new RegistrationException("Age cannot be null or empty");
+        }
```

Comment:
Consider combining the null checks for login, password, and age into a single condition to make the code more concise. For example, you can use `if (user.getLogin() == null || user.getPassword() == null || user.getAge() == null)`. This will reduce redundancy and improve readability.